### PR TITLE
fix: [H01] fixes duplicate request rewards

### PR DESCRIPTION
**Pull request by md0x**
Signed-off-by: Pablo Maldonado <pablo@umaproject.org>

**Motivation**

```
The request processing logic within _updateAccountSlashingTrackers is not
able to correctly handle multiple rolled votes that appear sequentually in the request array.
```

**Summary**

To avoid adding more complexity to this logic, I implemented the simple change suggested by the audit.

```
A straightforward fix that sets requestIndex =
deletedRequests[requestIndex] on line 845 (remove the + 1 ) and then inserts a
continue statement. This would return execution back to line 843 where
requestIndex would be incremented as before, but ensures that the
deletedRequests[requestIndex] != 0 test on line 845 is applied to this new
index.
```

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

**Review by mrice32 — COMMENTED** [review]
@md0x Minor comment on the PR description:

**OZ Audit Fixes**

1. The title of audit fix PRs should be fix: [H01] some description of fix.
2. The description should include the exact text from the issue that is being fixed.
3. The description should include how you're planning to resolve the issue. Sometimes this is just following the suggestion of OZ, sometimes, it's more involved than that.
Examples: https://github.com/UMAprotocol/protocol/pull/3694, https://github.com/UMAprotocol/protocol/pull/3698

**All changes**

All PRs should have descriptions (even if short!) before being marked ready for review. This helps the reviewer understand the context for the change and helps others understand why the change was made when looking back in the history.

**Comment by md0x** [comment]
> @md0x Minor comment on the PR description:
> 
> **OZ Audit Fixes**
> 
> 1. The title of audit fix PRs should be fix: [H01] some description of fix.
> 2. The description should include the exact text from the issue that is being fixed.
> 3. The description should include how you're planning to resolve the issue. Sometimes this is just following the suggestion of OZ, sometimes, it's more involved than that. Examples: #3694, #3698
> 4. We should have a special github label for each audit so it's easy to search all fixes for that audit. I have created the `audit-dvm-2-0` label and added it to this PR.
> 
> **All changes**
> 
> All PRs should have descriptions (even if short!) before being marked ready for review. This helps the reviewer understand the context for the change and helps others understand why the change was made when looking back in the history.
> 
> It's a small thing, but doing it consistently really helps in the long-run!

Thanks @mrice32 for your comment and creating the label. I totally makes sense to me!

**Review by chrismaree — APPROVED** [review]
excellent. thank you for writing a detailed unit test as well. this looks good!

File: packages/core/test/oracle/VotingV2.js [hunk 1]
```diff
@@ -3024,4 +3024,88 @@ describe("VotingV2", function () {
...
+      price.toString()
+    );
+
+    await voting.methods.updateTrackers(account1).send({ from: account1 });
+    await voting.methods.updateTrackers(account2).send({ from: account1 });
+
+    // Both users should have the same lastRequestIndexConsidered.
+    assert.equal((await voting.methods.voterStakes(account1).call()).lastRequestIndexConsidered, 4);
+    assert.equal((await voting.methods.voterStakes(account2).call()).lastRequestIndexConsidered, 4);
+  });
```
Comment (mrice32) [inline packages/core/test/oracle/VotingV2.js:3110]: How does this validate the new behavior? Did something above revert before? Or were these indices wrong before?

**Review by mrice32 — APPROVED** [review]
Looks great!!

File: packages/core/test/oracle/VotingV2.js (hunk 1 above)
Comment (md0x) [reply packages/core/test/oracle/VotingV2.js:3110]: Hey @mrice32, yes this last assertion failed before the changes:
`assert.equal((await voting.methods.voterStakes(account2).call()).lastRequestIndexConsidered, 4);`

Explanation:
The consecutive rolled votes were not handled properly. The second person processing the trackers with a second rolled vote would enter [this break](https://github.com/UMAprotocol/protocol/blob/0860b59dbf1ca437d45176d4585de74f7f034740/packages/core/contracts/oracle/implementation/VotingV2.sol#L867) and that was the end of it, leaving the next trackers unprocessed resulting in different `lastRequestIndexConsidered` across users.

**Pull request merged**